### PR TITLE
[Discovery]--create and update of configuration using config json file.

### PR DESCRIPTION
**Pull request by rahulmukherjee85**
Hi There,
I have implemented the create and update configuration feature  for discovery service. 
The configuration can done through a JSON file.

@
##### Checklist

- [X] `npm test` passes (tip: `npm run autofix` can correct most style issues)
- [X] tests are included
- [ ] documentation is changed or added
- [ ] link to public docs when adding new a service or new features for an existing service

##### New version_date Checklist

- [ ] A new constant is avaliable with the version_date - [example](https://github.com/watson-developer-cloud/node-sdk/blob/d1418ac2f9774194aaff0c8bd80f0d3722beef72/conversation/v1.js#L77)
- [ ] The new constant has a comment that summarizes the changes and/or links to relevant doc pages
- [ ] Any older version_date constants remain intact
- [ ] The error message thrown if the service is created without a version_date indicates the new version_date constant
- [ ] The example in the README includes the new version_date constant
- [ ] Any relevant code in the examples/ folder has been updated to use the new version_date constant
- [ ] Most tests are updated to the new version_date
- [ ] 1-2 new tests are added that use the old version_date (optional, but preferred)

### CODE REVIEW

**Comment by rahulmukherjee85** [comment]
Hello @nfriedly ,
Please have a look and let me know if any changes are required.
Thanks,
Rahul Mukherjee

File: discovery/v1.js [hunk 1]
```diff
@@ -177,6 +177,62 @@ DiscoveryV1.prototype.deleteEnvironment = function(params, callback) {
 };
 
 /**
+ * Creating a new configuration
```
Comment (germanattanasio) [inline discovery/v1.js]: Creates* instead of Creating

Comment (rahulmukherjee85) [inline discovery/v1.js]: Done.

**Comment by rahulmukherjee85** [comment]
Hello @germanattanasio ,
Please review, I have made the requested change.
Thanks,
Rahul Mukherjee.

File: discovery/v1.js [hunk 2]
```diff
@@ -177,6 +177,62 @@ DiscoveryV1.prototype.deleteEnvironment = function(params, callback) {
...
+      path: pick(params, ['environment_id']),
+      multipart: [
+        {
+          'content-type': 'application/json',
+          body: params.file
+        }
+      ],
+      json: true
+    },
+    requiredParams: ['environment_id'],
```
Comment (germanattanasio) [inline discovery/v1.js]: should `file` be a required parameter?

File: discovery/v1.js [hunk 3]
```diff
@@ -177,6 +177,62 @@ DiscoveryV1.prototype.deleteEnvironment = function(params, callback) {
...
+      path: pick(params, ['environment_id', 'configuration_id']),
+      multipart: [
+        {
+          'content-type': 'application/json',
+          body: params.file
+        }
+      ],
+      json: true
+    },
+    requiredParams: ['environment_id', 'configuration_id'],
```
Comment (germanattanasio) [inline discovery/v1.js]: should `file` be a required parameter?

**Comment by rahulmukherjee85** [comment]
Hello @germanattanasio,
Yes. File is a required parameter. Please have a look , I have made the requested change.
Regards,
Rahul Mukherjee.

**Comment by rahulmukherjee85** [comment]
Thanks !!!

**Pull request merged**